### PR TITLE
#139 different colors for different events types in calendar

### PR DESCRIPTION
**Pull request by andrewMuntyan**

### CODE REVIEW

File: client/static/app/modules/calendar/services/calendar.service.js [hunk 1]
```diff
@@ -32,11 +32,13 @@ define(function(require) {
     }
 
     function transformPlayToCalendar(data) {
+      console.log(data);
```
Comment (artyomtrityak) [inline client/static/app/modules/calendar/services/calendar.service.js]: log

File: client/static/app/modules/play/controllers/plays-list.controller.js [hunk 2]
```diff
@@ -33,11 +33,12 @@ define(function(require) {
         vm.pageChanged = pageChanged;
         vm.onJoinClick = onJoinClick;
         vm.reloadList = reloadList;
-        vm.tillNow = tillNow;
         vm.playsCount = playsCount.data.count;
         vm.descriptionLimit = 150;
         vm.isLoggedIn = dgUserService.isLoggedIn();
 
+      console.log(vm);
```
Comment (artyomtrityak) [inline client/static/app/modules/play/controllers/plays-list.controller.js]: log

File: client/static/app/modules/calendar/services/calendar.service.js (hunk 1 above)
Comment (wsite) [inline client/static/app/modules/calendar/services/calendar.service.js]: remove?

File: server/modules/plays/model.js [hunk 3]
```diff
@@ -50,9 +50,10 @@ var PlaySchema = new Schema({
     type: Schema.Types.ObjectId,
     ref: 'Users'
   },
-  status: {
-    type: 'string',
-    enum: ['not started', 'ended', 'canceled']
+  _canceled: {
+    type: Boolean
+//    required: true
```
Comment (artyomtrityak) [inline server/modules/plays/model.js]: remove it

File: server/modules/plays/model.js [hunk 4]
```diff
@@ -61,30 +62,53 @@ PlaySchema.statics.getPlays = getPlays;
...
+
 module.exports = mongoose.model('Plays', PlaySchema);
 
+function tillNow(targetDate){
+  targetDate = new Date() instanceof Date ? targetDate : new Date(targetDate);
+  return targetDate.getTime() <= (new Date()).getTime();
+}
+
 function getPlays(startDate, endDate, city, page, search, onlyMy, includeOld, userId) {
+  console.log('');
```
Comment (artyomtrityak) [inline server/modules/plays/model.js]: logs

File: server/modules/plays/model.js [hunk 5]
```diff
@@ -61,30 +62,53 @@ PlaySchema.statics.getPlays = getPlays;
...
+  if (city) {
+    queryObj['city.id'] = city;
+  }
 
   if (onlyMy) {
     queryObj.creator = userId;
   }
 
+
+  console.log('');
```
Comment (artyomtrityak) [inline server/modules/plays/model.js]: logs

File: server/modules/plays/model.js [hunk 6]
```diff
@@ -61,30 +62,53 @@ PlaySchema.statics.getPlays = getPlays;
...
+    return this.canceled ? 'canceled' : (tillNow(this.when) ? 'ended' : 'not-started');
+  });
+
+PlaySchema.set('toJSON', { virtuals: true });
+
 module.exports = mongoose.model('Plays', PlaySchema);
 
+function tillNow(targetDate){
+  targetDate = new Date() instanceof Date ? targetDate : new Date(targetDate);
+  return targetDate.getTime() <= (new Date()).getTime();
```
Comment (artyomtrityak) [inline server/modules/plays/model.js]: you can use `Date.now()`

**Comment by artyomtrityak** [comment]
merge

**Pull request merged**